crates/cranelift/src/debug/transform/debug_transform_logging.rs

```diff
@@ -0,0 +1,241 @@
+use crate::debug::Reader;
+use core::fmt;
+use gimli::{write, AttributeValue, DebuggingInformationEntry, Dwarf, LittleEndian, Unit};
+
+macro_rules! dbi_log {
+    ($($tt:tt)*) => {
+        #[cfg(debug_assertions)]
+        ::log::trace!(target: "debug-info-transform", $($tt)*);
```

Comment:
This a placeholder for what should ideally be `if #cfg(feature = "trace-log")`, same as Cranelift's `trace!`, however, I do not know how to properly encode this into the build files.

Reply:
The end result in terms of user experience that I would like to see here is:

1) This logging is enabled by default for debug builds.
2) This logging _can_ be enabled for release builds.
3) There is a "one stop shop" feature for all verbose logging.

Notably, AFAICT, Cranelift's `trace!` does not work for `1`, for which presumably there is a good reason. It means that `1` and `3` are in conflict. Would you say `1` should win (implying adding a separate feature just to this crate)?

Reply:
I generally agree with these goals; and to fill in a bit of detail, `trace!` is off by default (compiled out statically) when the option is disabled because in release settings it causes unacceptable overheads to even have the dynamic check/branch on the logging level in inner loops. This carries over to debug builds because we haven't done anything special to enable the option in debug builds.

I think we can get all three (i.e. I don't see any conflicts) by:
- Putting this crate's logging macro, and `cranelift-codegen`'s, in code that is enabled with something like `#![cfg(any(feature = "trace-log", debug_assertions))]`;
- Adding a `trace-log` feature to the `wasmtime` and `wasmtime-cli` crates (off by default) that in turn enables `trace-log` on `wasmtime-cranelift` and on `cranelift-codegen`.

Does this seem reasonable to you?

Reply:
Sounds good, I'll take a look at implementing it.